src/main/java/org/mockito/internal/util/MockUtil.java

```diff
@@ -43,26 +43,29 @@ public static TypeMockability typeMockabilityOf(Class<?> type) {
     }
 
     public static <T> void resetMock(T mock) {
-        InternalMockHandler oldHandler = (InternalMockHandler) getMockHandler(mock);
+        MockHandler oldHandler = getMockHandler(mock);
         MockCreationSettings settings = oldHandler.getMockSettings();
         MockHandler newHandler = createMockHandler(settings);
 
         mockMaker.resetMock(mock, newHandler, settings);
     }
 
-    public static <T> InternalMockHandler<T> getMockHandler(T mock) {
+    public static <T> MockHandler<T> getMockHandler(T mock) {
         if (mock == null) {
             throw new NotAMockException("Argument should be a mock, but is null!");
         }
 
         if (isMock(mock)) {
-            MockHandler handler = mockMaker.getHandler(mock);
-            return (InternalMockHandler) handler;
+            return mockMaker.getHandler(mock);
         } else {
             throw new NotAMockException("Argument should be a mock, but is: " + mock.getClass());
         }
     }
 
+    public static InvocationContainerImpl getInvocationContainer(Object mock) {
+        return (InvocationContainerImpl) getMockHandler(mock).getInvocationContainer();
```

Comment:
I think that InvocationContainer cannot be practically made public. It's too coupled with Mockito impl. details. I wrote details in the commit message and high level information in the Javadoc.

Do you have suggestions how to make it better?